devices/src/legacy/rtc_pl031.rs

```diff
@@ -174,7 +172,9 @@ impl BusDevice for RTC {
 
     fn write(&mut self, offset: u64, data: &[u8]) {
         if data.len() <= 4 {
-            let v = LittleEndian::read_u32(data);
+            let mut array = [0; 4];
+            array.copy_from_slice(data);
```

Comment:
This panics if `data` has less than 4 bytes.

Reply:
`LittleEndian::read_32(data)` also results in a panic when `data` is less than 4 bytes. In the scenario when that occurs, should we pad `data` with 0s to bring it up to 4 bytes?

Reply:
Nice catch! Yes, the right thing to do is pad.